### PR TITLE
Elliot sga budget details

### PR DESCRIPTION
**Pull request by Elliot-TS**
## Why / Feature

https://trello.com/c/392TFgHL
https://trello.com/c/Y9XSoYKx
https://trello.com/c/pKWUX326

### Comments
- Created Comment data structure and edited Budget and Item data structures to allow for comments
- Added comments collection in firebase
- Added methods to DataModel to interact with comments

### Budget Views
- Created an SGA Budget View
- SGA can Approve or Deny budget
- SGA can deny individual items
- RSO can see what items were denied once the SGA submits the denial

## How can I check if it works?

Log in as an SGA user and click on a submitted budget in the dashboard to navigate to the Budget View page

## How did I test?

I added sgaBudgetview.test
I wrote draft tests for comments, but I removed 'test' from the filename because the tests did not work, and the back-end comment code isn't yet used by the front-end.

### CODE REVIEW

File: git
Comment (tandetat) [inline git:1]: Make sure to remove this file before merging

**Review by SimonMTaye — COMMENTED** [review]
Clicking the deny budget button should redirect to the dashboard. Right now, it doesn't refresh or redirect the user back to the dashboard. We should refresh the dashboard by calling `revalidate(/dashboard)`  and `redirect(/dashboard)`

File: test/app/sgabudgetview.test.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    expect(item.current_status).toBe('denied');
+
+    // Undeny the first item
+    await user.click(itemDenyButtons[0]);
+    item = await mockDataprovider.getItem('denied_item');
+    expect(item.current_status).toBe('created');
+  });
+  /*
+
+  it('clicking deny budget button denies the whole budget', async () => {
```
Comment (tandetat) [inline test/app/sgabudgetview.test.tsx:113]: Why are these tests commented out?

Comment (Elliot-TS) [reply test/app/sgabudgetview.test.tsx:113]: Good catch!  I did that to test the previous ones one at a time, but I forgot to finish! ;)

**Review by SimonMTaye — APPROVED** [review]

**Pull request merged**